### PR TITLE
Restructuring of existing OGS below-ground concepts...

### PR DESCRIPTION
**Pull request by mcwimm**
... in preparation for a new bg concept (`NetworkOGSLargeScale3DExternal`)

The changes should not have touched the functionality, but only deal with merging/moving code and renaming functions and variables to be able to use the functionality in child classes and make the functions clearer.

Nonetheless, it would be really cool if someone would test the new concepts (maybe @jvollhueter ?).

_This PR is based on #112._

### CODE REVIEW

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py [hunk 1]
```diff
@@ -245,32 +280,52 @@ def copyPythonScript(self):
...
-        prefactors_filename = path.join(self._ogs_project_folder,
-                                        "salinity_prefactors.npy")
+
+        # OGS
+        try:
+            constants_filename = path.join(self._ogs_project_folder,
+                                           "constant_contributions.npy")
+            prefactors_filename = path.join(self._ogs_project_folder,
+                                            "salinity_prefactors.npy")
+        except:
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:294]: please specify except.

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py [hunk 2]
```diff
@@ -101,107 +90,37 @@ def calculateBelowgroundResources(self):
...
-            target.write(line)
-        source.close()
-        target.close()
+        self.renameParameters()
+
+    ## This function returns the directory of the python_source file in the
+    # directory of the concept if no external source file is provided.
+    def getSourceDir(self):
+        # ToDo: @JB: this is exactly the same function as in super() - how
+        #  can we avoid this but also get the directory of the concept
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py:124]: what do you mean?
what happens, if you use the inherited function?

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py [hunk 3]
```diff
@@ -141,10 +123,18 @@ def prepareNextTimeStep(self, t_ini, t_end):
         filename = path.join(
             self._ogs_project_folder,
             str(t_ini).replace(".", "_") + "_" + self._ogs_project_file)
-        self._tree.write(filename)
-        ## List containing reduction factor for each tree
+        self._tree.write(filename) # ToDo @JB: Was passiert hier?
+
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:127]: self._tree is the xml-tree from the ogs project file. 
above, the xm tree is updated (e.g. line 108). in line 127, the new project file for ogs is saved.

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py [hunk 4]
```diff
@@ -313,3 +368,33 @@ def writePVDCollection(self):
...
+            str(self._t_ini).replace(".", "_") + "_" + self._ogs_project_file)
+        print("Running ogs...")
+        bc_path = (path.dirname(path.dirname(path.abspath(__file__))))
+
+        if not (os.system(bc_path + "/OGS/bin/ogs " + current_project_file +
+                          " -o " + self._ogs_project_folder + " -l error")
+                == 0):
+            raise ValueError("Ogs calculation failed!")
+        print("OGS-calculation done.")
+        self.writePVDCollection()
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:384]: writepvdcollection is creating a pvd file (for animations in paraview) which is collecting the groundwater states at the end of ogs simulations.
example: you simulate for 10 years whith timesteps of length = 1/2 year. the pvd collection will then contain references to the porewater salinity distribution after 0, 0.5, 1, ... years.

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py (hunk 1 above)
Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py]: I removed try-except as it had no effect there.

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py (hunk 2 above)
Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py]: If we inherit the function, it still uses the path of super().

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py (hunk 4 above)
Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:378]: and below? `for file in files`

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py (hunk 3 above)
Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:127]: added

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py (hunk 2 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py]: i have no workaround. so just leave it as it is?

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py (hunk 4 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:382]: schreibe zeilenweise alle filenames in das pvd-file

File: TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py (hunk 2 above)
Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/NetworkOGSLargeScale3D/NetworkOGSLargeScale3D.py]: yes, i agree.

File: TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py (hunk 4 above)
Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/OGSLargeScale3D/OGSLargeScale3D.py:382]: ok. I moved those lines to the method writePVD...

**Pull request merged**